Makefile

```diff
@@ -315,13 +315,74 @@ src/newrelic/infinite_tracing/com_newrelic_trace_v1/v1.pb.go: protocol/infinite_
 #
 
 .PHONY: integration
-integration: Makefile daemon lasp-test-all
+integration: Makefile daemon lasp-test-all integration-events-limits
 	for PHP in $${PHPS:-8.1 8.0 7.4 7.3 7.2 7.1 7.0 5.6 5.5}; do \
           echo; echo "# PHP=$${PHP}"; \
 	  env NRLAMP_PHP=$${PHP} bin/integration_runner $(INTEGRATION_ARGS) || exit 1; \
 	  echo "# PHP=$${PHP}"; \
 	done
 
+#
+# Agent event limits integration testing
+#
+# Because of how the integration_runner connects to the collector once before
+# running any tests we have to tell the runner the value the agent would
+# have passed to it for each test.  This means these tests are not testing
+# the agent <-> daemon communcations of the agent's requested custom event limit
+# via the daemon to the collector and daemon sending back the collectors harvest
+# limit value.
+#
+
+.PHONY: integration-events-limits
+integration-events-limits: daemon
+	# create array with the collector response for each agent requested custom events max samples
+	# currently based on fast harvest cycle being 5 seconds so ratio is 12:1
+	declare -A custom_limits_tests; \
+	custom_limits_tests[240]=20; \
+	custom_limits_tests[7000]=583; \
+	custom_limits_tests[30000]=2500; \
+	custom_limits_tests[100000]=8333; \
+	for PHP in $${PHPS:-8.1 8.0 7.4 7.3 7.2 7.1 7.0 5.6 5.5}; do \
```

Comment:
Is this something we can extract out to a single place (function/variable)? It's added a lot here (and is already used extensively in the Makefile), but it would be nice to be able to configure the PHPS list in one place

Reply:
Good idea - I'd prefer not to do that refactoring in this particular PR but maybe we can add it to a list of build/CI/CD improvements we want to do?